programs/ziti-edge-tunnel/ziti-edge-tunnel.c

```diff
@@ -1129,6 +1130,19 @@ static void on_event(const base_event *ev) {
     }
 }
 
+static char* normalize_host(char* hostname) {
+    size_t len = strlen(hostname);
+    char* hostname_new = calloc(len+1, sizeof(char));
+    // add . in the beginning of the hostname
+    if (hostname[len-1] == ".") {
```

Comment:
comparing `char` which is an integer type to a string (denoted by double quotes) is not valid, and will not work the way you are probably expecting. CLIon and probably any halfway decent IDE will warn you about this:
![image](https://user-images.githubusercontent.com/13525299/149955212-da0efe64-f319-4c94-8ca6-46c741f12a4c.png)

In this case you want to use single-quotes for your constant:
```
if (hostname[len-1] == '.') {
```